"Target Size Enhanced" - atomic rule (#2279)
auto-wcag/auto-wcag: pages/glossary/can-be-targeted-by-pointer-event.md

---
title: Can be target by a pointer event
key: can-be-targeted-by-pointer-event
unambiguous: true
objective: true
input_aspects:
  - CSS styling
  - DOM tree
---

An element _can be targeted by pointer events_ when all the following conditions are true:

- the element is a [semantic `widget`][semantic role]; and
- the element is [focusable][]; and
- the element [computed value][] of the `pointer-events` CSS property is `auto`; and

> - through scrolling, it is possible to have parts of the element's [border box][] which intersect the viewport and are not entirely covered by the [border boxes][] of elements with greater [computed][computed value] `z-index` and a [computed][computed value] `pointer-events` of `auto`.

Reviewer: ```suggestion
- through scrolling, it is possible to have parts of the element's [border box][] which intersect the viewport and do not overlap [border boxes][] of elements with greater [computed][computed value] `z-index` and a [computed][computed value] `pointer-events` of `auto`.
```

Perhaps it's more objective?

Author: I see the problem. I'm not sure this formulation is much better since the point it also about `z-index` and the like (if the element is totally covered by others, it is not clickable).

Author: I've rephrased it in terms of intersections. Also referring directly to CSS "panting order" definition since just looking at the z-index is far from enough 🙈

#### Background

> This definition tries to capture which HTML elements can actually react to pointer events. It is not possible to have an exact definition of these for two main reasons:

Reviewer 2: ```suggestion
This definition tries to capture which HTML elements can actually respond to pointer events. It is not possible to have an exact definition of these for two reasons:
```

> - Sometimes, the element that handles the event is not the element that appear to react to it, but an ancestor (or descendant) capturing the event during propagation or bubbling. In the most extreme case, the `body` element of a page could be the only one with an event handler, acting differently depending on where the event actually occurred. In such a case, a button would be perceived by users as something that can be targeted by a pointer event, while technically it is the `body` element which is targeted.

Reviewer: Reading this (and the next) reason, I was wondering if we should name this definition as "is perceived as a pointer event target"?

Author: I went for "Observed as a pointer events target". I felt "perceived" was having a too strong meaning, especially in our contexts.

> - Sometimes, the element that handles the event is not the element that appear to react to it, but an ancestor (or descendant) capturing the event during propagation or bubbling. In the most extreme case, the `body` element of a page could be the only one with an event handler, acting differently depending on where the event actually occurred. In such a case, a button would be perceived by users as something that can be targeted by a pointer event, while technically it is the `body` element which is targeted.

Reviewer 2: ```suggestion
- Sometimes, the element that handles the event is not the element that appears to react to it, but an ancestor (or descendant) capturing the event during propagation or bubbling. In the most extreme case, the `body` element of a page could be the only one with an event handler, acting differently depending on where the event actually occurred. In such a case, a button would be perceived by users as something that can be targeted by a pointer event, while technically it is the `body` element which is targeted.
```

> - It is not possible to query the list of event listeners on a given elements. Some User Agents offer way to monitor events fired at a given element, but none offer a way to query for event listeners. Additionally, an event listener might ultimately do nothing and thus, for users, the corresponding element wouldn't look like it can be targeted by pointer events (since it effectively wouldn't react to them).

Reviewer 2: ```suggestion
- It is not possible to query the list of event listeners on a given element. Some User Agents offer ways to monitor events fired at a given element, but none offer a way to query for event listeners. Additionally, an event listener might ultimately do nothing and thus, for users, the corresponding element wouldn't look like it could be targeted by pointer events (since it effectively wouldn't react to them).
```
